src/main/java/mate/academy/springbootintro/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+package mate.academy.springbootintro.service;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootintro.dto.OrderItemDto;
+import mate.academy.springbootintro.dto.OrderRequest;
+import mate.academy.springbootintro.dto.OrderDto;
+import mate.academy.springbootintro.dto.UpdateOrderStatusRequest;
+import mate.academy.springbootintro.exception.EntityNotFoundException;
+import mate.academy.springbootintro.mapper.OrderItemMapper;
+import mate.academy.springbootintro.mapper.OrderMapper;
+import mate.academy.springbootintro.model.Order;
+import mate.academy.springbootintro.model.ShoppingCart;
+import mate.academy.springbootintro.model.User;
+import mate.academy.springbootintro.model.CartItem;
+import mate.academy.springbootintro.model.OrderItem;
+import mate.academy.springbootintro.repository.cartitem.CartItemRepository;
+import mate.academy.springbootintro.repository.order.OrderRepository;
+import mate.academy.springbootintro.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.springbootintro.repository.user.UserRepository;
+import org.springframework.stereotype.Service;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepostiory;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, OrderRequest request) {
+        User user = userRepository.findById(userId)

```

Comment:
Use `getReferenceById` so you don't need additional optional handling. This comment applies to all cases below.